### PR TITLE
Velocity snapping hud

### PR DESCRIPTION
**Pull request by suburbski**

### CODE REVIEW

File: src/cgame/cg_timerun_draw.c [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include "cg_local.h"
+#include "..\game\q_shared.h"
```
Comment (boutetnico) [inline src/cgame/cg_timerun_draw.c]: Remove this, `q_shared.h` is already included in `cg_local.h`.

File: src/cgame/cg_timerun_draw.c [hunk 2]
```diff
@@ -856,7 +931,7 @@ void CG_DrawBannerPrint(void) {
 void CG_UpdateJumpSpeeds(void) {
 	int speed = 0;
 
-	if (cg.timerunJumpCounter < cg.predictedPlayerState.identifyClientHealth && cg.timerunJumpCounter < (int)(sizeof (cg.timerunJumpSpeeds) / sizeof (cg.timerunJumpSpeeds[0]))) {
+	if (cg.timerunJumpCounter < cg.predictedPlayerState.identifyClientHealth && cg.timerunJumpCounter < (int)(sizeof(cg.timerunJumpSpeeds) / sizeof(cg.timerunJumpSpeeds[0]))) {
```
Comment (boutetnico) [inline src/cgame/cg_timerun_draw.c]: `sizeof` is a preprocessor directive not a C function. It should be followed by a space to highlight the difference with any C function.

**Comment by boutetnico** [comment]
Can we have UI menu to control this and keep consistency with other cvars?

File: src/cgame/cg_timerun_draw.c (hunk 2 above)
Comment (suburbski) [reply src/cgame/cg_timerun_draw.c]: well done visual assist

**Comment by boutetnico** [comment]
I think we should disable this when `g_cupMode` is enabled. As a reminder, `g_cupMode` a specific server-side setting, that forces players to run game at certain settings like player hiding, no CGaz, etc...

**Comment by boutetnico** [comment]
@suburbski please rebase

**Comment by boutetnico** [comment]
This PR creates a compiler warning on MacOS with clang:

```
[ 28%] Building C object CMakeFiles/cgame.dir/src/cgame/cg_timerun_draw.c.o
/Users/nicolas/Code/et/ETrun/ETrun/src/cgame/cg_timerun_draw.c:699:73: warning: comparison of integers
      of different signs: 'int' and 'unsigned long' [-Wsign-compare]
  ...floor(cg.snapSpeed + 0.5) - 0.5; step > 0 && cg.snapCount < sizeof(cg.snapZones) - 2; step--) {
                                                  ~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

**Pull request merged**